File: algorithms/0169.majority-element/solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn majority_element(nums: Vec<i32>) -> i32 {
        let mut res = 0i32;
        let mut map: HashMap<i32, i32> = HashMap::new();
        for n in nums {
            *map.entry(n).or_default() += 1;

            if map.get(&n) > map.get(&res) {
                res = n;
            }
        }

        res
    }
}

File: algorithms/0169.majority-element/solution.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int res;
        unordered_map<int, int> m;

        for (int n : nums) {
            m[n]++;

            if (m[n] > m[res]) {
                res = n;
            }
        }

        return res;
    }
};
